src/controllers/services.js

```diff
@@ -0,0 +1,101 @@
+import { fetchCollections } from '../services/collections.js';
+import { fetchServices } from '../services/services.js';
+
+export const getServices = async (req, res) => {
+  const { name, termsTypes } = req.query;
+
+  const collections = await fetchCollections().catch(error => {
+    res.status(error.response?.status || 500).json({ error: `${error.message} (${error.code})` });
+  });
+
+  if (!collections) {
+    return;
+  }
+
+  const results = [];
+  const failures = [];
+
+  await Promise.all(collections.map(async ({ id: collectionId, endpoint }) => {
+    const services = await fetchServices(endpoint).catch(error => {
+      failures.push({ collection: collectionId, message: error.toString() });
+    });
+
+    if (!services) {
+      return;
+    }
+
+    for (const service of services) {
+      if (!service.terms || (name && service.name !== name)) {
+        continue;
+      }
```

Comment:
⚡ 🌱 Using `find` would on average cut half of the processing power used for iterating over all services here.

> :brain: In general, I would personally find a functional approach way more readable.
```suggestion
    if (name) {
      services = [ services.find(service => service.name == name) ];
    }
    services = services.filter(service => service.terms);
```